hypervisor/src/kvm/mod.rs

```diff
@@ -1602,6 +1619,15 @@ impl cpu::Vcpu for KvmVcpu {
     fn set_immediate_exit(&self, exit: bool) {
         self.fd.set_kvm_immediate_exit(exit.into());
     }
+
+    ///
+    /// Returns a structure with details about TDX exit reason
+    ///
+    #[cfg(feature = "tdx")]
+    fn get_tdx_exit_details(&mut self) -> TdxExitDetails {
```

Comment:
Can't you interpret the values here and turn them into an enum. So you don't need to leak the kvm_run struct